### PR TITLE
Add Submissions

### PR DESCRIPTION
**Pull request by owlester12**

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -117,4 +117,4 @@
   "engines": {
     "node": "^20.0.0"
   }
-}
\ No newline at end of file
+}
```
Comment (julianweng) [inline frontend/package.json:120]: Is this necessary to pass lint?

File: frontend/pages/settings.tsx [hunk 2]
```diff
@@ -31,11 +35,18 @@ const Notification = styled.span`
 `
 
 type SettingsProps = {
-  userInfo: UserInfo
+  userInfo?: UserInfo
   authenticated: boolean | null
+  submissions: ApplicationSubmission[]
+  whartonapplications: any
```
Comment (julianweng) [inline frontend/pages/settings.tsx:41]: Nit: camel case variable name and adding types would be nice (I know the reference code didn't have either but it would be good practice)

File: frontend/pages/settings.tsx [hunk 3]
```diff
@@ -68,6 +79,16 @@ const Settings = ({ userInfo, authenticated }: SettingsProps) => {
...
     },
+    {
+      name: 'submissions',
+      label: 'Submissions',
+      content: <SubmissionsPage initialSubmissions={submissions} />,
+    },
+    {
+      name: 'applications',
+      label: 'Applications',
+      content: <ApplicationsPage whartonapplications={whartonapplications} />,
```
Comment (julianweng) [inline frontend/pages/settings.tsx:90]: Now that the page is more visible in the profile, would be nice to clarify that it's for Wharton Cycle applications specifically (maybe change it to all eventually but not needed now).

**Review by julianweng — CHANGES_REQUESTED** [review]
Looks good generally, had a couple of nits and things that could be changed from the existing ported implementation. Thanks for taking the time to revisit this!

File: frontend/package.json (hunk 1 above)
Comment (owlester12) [reply frontend/package.json:120]: Yes lmao

File: frontend/components/Applications.tsx [hunk 4]
```diff
@@ -103,41 +103,44 @@ function ApplicationsPage({ whartonapplications }): ReactElement {
...
-                          __html: application.description,
-                        }}
-                      ></DescriptionWrapper>
-                    )}
-                </Card>
-              </Link>
-            </CardWrapper>
-          ))
+          <div>
+            <Text>Only Wharton applications displayed here!</Text>
```
Comment (julianweng) [inline frontend/components/Applications.tsx:107]: Maybe rephrase as "Note: only current Wharton applications are displayed on this page"

File: frontend/pages/settings.tsx [hunk 5]
```diff
@@ -87,7 +87,7 @@ const Settings = ({
     {
       name: 'applications',
       label: 'Applications',
-      content: <ApplicationsPage whartonapplications={whartonapplications} />,
+      content: <ApplicationsPage whartonapplications={whartonApplications} />,
```
Comment (julianweng) [inline frontend/pages/settings.tsx:90]: Could be appropriate to also rename the prop in ApplicationsPage to be camelcase for simplicity

**Review by julianweng — CHANGES_REQUESTED** [review]
Thanks for taking the time to fix the nits, just left two more and it should be ready!

File: frontend/components/Applications.tsx (hunk 4 above)
Comment (owlester12) [reply frontend/components/Applications.tsx]: Made changes

File: frontend/pages/settings.tsx (hunk 5 above)
Comment (owlester12) [reply frontend/pages/settings.tsx]: Made changes

**Review by julianweng — CHANGES_REQUESTED** [review]
Hi, thanks for making the changes, can you fix the type check errors (tsc) with your change? Also, there is probably a better way of handling the rename than declaring and immediately spreading a separate returner variable that would be clearer to read.

**Comment by owlester12** [comment]
Both issues should fixed. Some of the checks are still failing– something about the Codecov report not uploading? I'm unsure what that means.
